src/observability_utils/tracing/asserting_exporter.py

```diff
@@ -0,0 +1,56 @@
+from collections.abc import Callable, Sequence
+from concurrent.futures import Future
+from contextlib import contextmanager
+from typing import IO
+
+from opentelemetry.sdk.trace import ReadableSpan
+from opentelemetry.sdk.trace.export import (
+    SpanExporter,
+    SpanExportResult,
+)
+
+
+class JsonObjectSpanExporter(SpanExporter):
+    """A custom span exporter to allow spans created by open telemetry tracing code to
+    be examined and verified during normal testing. When using the class it is
+    recommended to use the SimpleSpanProcessor rather than the BatchSpanProcessor to
+    keep the tests quick.
+    """
+
+    def __init__(
+        self,
+        service_name: str | None = "Test",
+        out: IO | None = None,
+        formatter: Callable[[ReadableSpan], str] | None = None,
+    ):
+        self.service_name = service_name
+        self.top_span: Future | None = None
+
+    def export(self, spans: Sequence[ReadableSpan]) -> SpanExportResult:
+        if self.top_span is None or self.top_span.done():
+            self.top_span = Future()
+            self.top_span.set_result(spans[-1])
+        return SpanExportResult.SUCCESS
+
+    def force_flush(self, timeout_millis: int = 30000) -> bool:
+        return True
+
+
```

Comment:
```suggestion


```

Since this implementation is the same as the super class, should be enough to shut it up.

Reply:
Not in fact true as demonstrated